pkg/volume/downwardapi/downwardapi.go

```diff
@@ -117,13 +117,8 @@ func (plugin *downwardAPIPlugin) NewUnmounter(volName string, podUID types.UID)
 }
 
 func (plugin *downwardAPIPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.Spec, error) {
-	downwardAPIVolume := &v1.Volume{
-		Name: volumeName,
-		VolumeSource: v1.VolumeSource{
-			DownwardAPI: &v1.DownwardAPIVolumeSource{},
-		},
-	}
-	return volume.NewSpecFromVolume(downwardAPIVolume), nil
+	//  To reconstruct volume spec we need source name which cannot be retrieved from volumeName and mountPath.
+	return nil, fmt.Errorf("impossible to reconstruct volume spec from volume mountpath")
```

Comment:
Can you elaborate here? What do you mean by source name? I assume it is same as `outerVolumeSpecName`  but I think we need to document this better, why current values were wrong.